plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/expression/ImplementAvg.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.pinot.query.expression;
+
+import io.trino.matching.Capture;
+import io.trino.matching.Captures;
+import io.trino.matching.Pattern;
+import io.trino.plugin.pinot.PinotColumnHandle;
+import io.trino.spi.connector.AggregateFunction;
+import io.trino.spi.expression.Variable;
+
+import java.util.Optional;
+
+import static io.trino.matching.Capture.newCapture;
+import static io.trino.plugin.pinot.query.DynamicTableBuilder.SUPPORTED_INPUT_TYPES;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.basicAggregation;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.expressionType;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.functionName;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.singleInput;
+import static io.trino.plugin.pinot.query.expression.AggregateFunctionPatterns.variable;
+import static java.lang.String.format;
+
+public class ImplementAvg
+        implements AggregateFunctionRule
+{
+    // Extracted from io.trino.plugin.jdbc.expression
+    private static final Capture<Variable> INPUT = newCapture();
+
+    @Override
+    public Pattern<AggregateFunction> getPattern()
+    {
+        return basicAggregation()
+                .with(functionName().equalTo("avg"))
+                .with(singleInput().matching(
+                        variable()
+                                .with(expressionType().matching(SUPPORTED_INPUT_TYPES::contains))
```

Comment:
SUPPORTED_INPUT_TYPES doesn't look like specific to averages, so reuse here looks wrong.

i suggest defining supported types directly here, in `ImplementAvg`

(some for other cases)

Reply:
@elonazoulay bumping this up again

Reply:
I moved SUPPORTED_INPUT_TYPES to ImplementAvg, I think that line is unchanged, but above it's declared private static, no longer imported.